crypto/rand/rand_lib.c

```diff
@@ -185,6 +185,10 @@ static int setup_drbg(RAND_DRBG *drbg)
     drbg->lock = CRYPTO_THREAD_lock_new();
     ret &= drbg->lock != NULL;
     drbg->size = RANDOMNESS_NEEDED;
+    drbg->secure = CRYPTO_secure_malloc_initialized();
+    drbg->randomness = drbg->secure
+        ? OPENSSL_secure_malloc(rand_bytes.size)
+        : OPENSSL_malloc(rand_bytes.size);
     drbg->randomness = OPENSSL_malloc(drbg->size);
```

Comment:
Double `malloc`. I suppose this line should be removed?
Should the size for the new malloc calls also be `drbg->size`?